benchmark/Streamly/Benchmark/Prelude/Serial/Elimination.hs

```diff
@@ -0,0 +1,683 @@
+-- |
+-- Module      : Serial.Elimination
+-- Copyright   : (c) 2018 Composewell Technologies
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE RankNTypes #-}
+
+#ifdef __HADDOCK_VERSION__
+#undef INSPECTION
+#endif
+
+#ifdef INSPECTION
+{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_GHC -fplugin Test.Inspection.Plugin #-}
+#endif
+
+module Serial.Elimination (benchmarks) where
+
+import Control.DeepSeq (NFData(..))
+import Data.Functor.Identity (Identity, runIdentity)
+import System.Random (randomRIO)
+
+import qualified Data.Foldable as F
+import qualified GHC.Exts as GHC
+
+#ifdef INSPECTION
+import GHC.Types (SPEC(..))
+import Test.Inspection
+
+import qualified Streamly.Internal.Data.Stream.StreamD as D
+#endif
+
+import qualified Streamly.Prelude  as S
+import qualified Streamly.Internal.Data.Stream.IsStream as Internal
+
+import Gauge
+import Streamly.Prelude (SerialT, IsStream, serially)
+import Streamly.Benchmark.Common
+import Streamly.Benchmark.Prelude
+import Prelude hiding (length, sum, or, and, any, all, notElem, elem, (!!),
+    lookup, repeat, minimum, maximum, product, last, mapM_, init)
+import qualified Prelude
+
+{-# INLINE repeat #-}
+repeat :: (Monad m, S.IsStream t) => Int -> Int -> t m Int
```

Comment:
This should go under generation.

Reply:
It is in generation as well. It is repeated here only because `the` needs it.